### PR TITLE
Use backing file for storing validated block cids

### LINKED ISSUE #2642: Remove block validation meta-data from the blockstore
**Issue summary**

Forest keeps track of which blocks have been validated by storing a set of validated CIDs in the blockstore. In the future, we want the blockstore to only contain the blockchain itself. The set of validated blocks should therefore be moved somewhere else.

The store of validated blocks does not need to be crash proof. Losing data just means more blocks will be re-validated. We could use an approach similar to #2635 but perhaps only sync the data to disk once every 10 minutes.

It would be tempting to get rid of the set of validated blocks and go by the heaviest tipset. This would require significant changes to the code base, though.

**Other information and links**

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:
- Use backing file for storing validated blocks instead of store
- Fix typo in documentation

## Reference issue to close (if applicable)

Closes #2642

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Review by lemmih — CHANGES_REQUESTED** [review]
A lot of `Store` constraints can be removed from the rest of the code now. Let's remove as many as possible.

**Comment by lemmih** [comment]
Oh, and the checklist needs to be completed.

File: utils/forest_utils/src/db/file_backed_obj.rs [hunk 1]
```diff
@@ -1,36 +1,61 @@
...
 use cid::Cid;
 use log::warn;
 
 pub struct FileBacked<T: FileBackedObject> {
     inner: T,
     path: PathBuf,
+    last_sync: Option<SystemTime>,
 }
 
+const SYNC_PERIOD: Duration = Duration::from_secs(600);
```
Comment (LesnyRumcajs) [inline utils/forest_utils/src/db/file_backed_obj.rs:20]: Shall we add it to the configuration?

File: utils/forest_utils/src/db/file_backed_obj.rs [hunk 2]
```diff
@@ -1,36 +1,61 @@
...
+        if let Some(last_sync) = self.last_sync {
+            if now.duration_since(last_sync)? > SYNC_PERIOD {
+                self.last_sync = Some(now);
+                self.flush_to_file()?;
+            }
+            return Ok(());
+        }
+        self.last_sync = Some(now);
+        Ok(())
+    }
```
Comment (LesnyRumcajs) [inline utils/forest_utils/src/db/file_backed_obj.rs:50]: Fancy adding a unit test for this method?

File: utils/forest_utils/src/db/file_backed_obj.rs (hunk 1 above)
Comment (elmattic) [reply utils/forest_utils/src/db/file_backed_obj.rs:20]: I'm not sure if it's really critical for our users to be configurable. I prefer to leave it like this for now (ten minutes should be a good start).

File: utils/forest_utils/src/db/file_backed_obj.rs (hunk 2 above)
Comment (elmattic) [reply utils/forest_utils/src/db/file_backed_obj.rs:50]: Let me think.

File: utils/forest_utils/src/db/file_backed_obj.rs (hunk 1 above)
Comment (LesnyRumcajs) [reply utils/forest_utils/src/db/file_backed_obj.rs:20]: If you make it configurable then you will be able to add some unit tests. :)

Comment (LesnyRumcajs) [reply utils/forest_utils/src/db/file_backed_obj.rs:20]: But modifying `last_sync` directly is also fine.

Comment (elmattic) [reply utils/forest_utils/src/db/file_backed_obj.rs]: I've made it configurable.

File: utils/forest_utils/src/db/file_backed_obj.rs (hunk 2 above)
Comment (elmattic) [reply utils/forest_utils/src/db/file_backed_obj.rs]: Added two tests.

File: utils/genesis/src/lib.rs [hunk 3]
```diff
@@ -145,7 +145,7 @@ where
     // Update head with snapshot header tipset
     sm.chain_store().set_heaviest_tipset(ts.clone())?;
 
-    sm.blockstore().flush()?;
+    //sm.blockstore().flush()?;
```
Comment (lemmih) [inline utils/genesis/src/lib.rs:148]: We still need this flush, unfortunately.

Comment (elmattic) [reply utils/genesis/src/lib.rs:148]: Will add back.

File: vm/state_migration/src/nv12/miner.rs [hunk 4]
```diff
@@ -13,7 +13,7 @@ use cid::multihash::Code::Blake2b256;
 use cid::Cid;
 use fil_actor_interface::actorv3::miner::State as V3State;
 use fil_actor_interface::actorv4::miner::State as V4State;
-use forest_db::Store;
+
```
Comment (lemmih) [inline vm/state_migration/src/nv12/miner.rs:16]: spaces?

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**